tests/testnet/network.go

```diff
@@ -326,32 +315,32 @@ func (n *testNetwork) getMessageDeliveryTransactionReceipt(
 		startBlock = 0
 	}
 
-	abi, err := teleportermessenger.TeleporterMessengerMetaData.GetAbi()
-	if err != nil {
-		return nil, err
-	}
-
 	// Get the log event of the delivery. The log must be in the last {receiveCrossChainMessageLookBackBlocks} blocks.
-	logs, err := destination.RPCClient.FilterLogs(ctx, subnetevminterfaces.FilterQuery{
-		FromBlock: big.NewInt(int64(startBlock)),
-		Addresses: []common.Address{n.teleporterContractAddress},
-		Topics: [][]common.Hash{
-			{abi.Events[receiveCrossChainMessageEventName].ID},
-			{common.BytesToHash(teleporterMessageID[:])},
-			{common.BytesToHash(sourceBlockchainID[:])},
+	logIter, err := destination.TeleporterMessenger.FilterReceiveCrossChainMessage(
+		&bind.FilterOpts{
+			Start:   startBlock,
+			Context: cctx,
 		},
-	})
+		[][32]byte{teleporterMessageID},
+		[][32]byte{sourceBlockchainID},
+		nil,
+	)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(logs) == 0 {
-		return nil, errors.New("Failed to find ReceiveCrossChainMessage log for relayed message")
-	} else if len(logs) > 1 {
-		return nil, errors.New("Found multiple ReceiveCrossChainMessage logs for relayed message")
+	// There should be exactly one matching event.
+	if !logIter.Next() {
```

Comment:
I like to change to use the abi bindings filter, might be worth adding a comment that `iter.Next()` populates the `Event` value and needs to be called first.